### PR TITLE
disk: remove expand auto snapshot

### PR DESCRIPTION
**Pull request by huww98**
#### What type of PR is this?

/kind bug
/kind deprecation

#### What this PR does / why we need it:

Following up #1182 , remove this function since it is not working and will block resizing.

preserve the function of deleting snapshot on successful resize for one more version, to avoid leaking snapshots.

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
DisableExpandAutoSnapshot feature gate is now locked to true, auto snapshot on resizing is no longer supported
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: pkg/features/features.go [hunk 1]
```diff
@@ -46,7 +46,7 @@ var (
 		DiskADController:           {Default: false, PreRelease: featuregate.Alpha},
 		DiskParallelAttach:         {Default: false, PreRelease: featuregate.Alpha},
 		DiskParallelDetach:         {Default: false, PreRelease: featuregate.Alpha},
-		DisableExpandAutoSnapshots: {Default: false, PreRelease: featuregate.Alpha},
```
Comment (mowangdk) [inline pkg/features/features.go:49]: Set the featuregate to default first? code remove in next version

Comment (huww98) [reply pkg/features/features.go:49]: please see #1220

**Comment by huww98** [comment]
Do you want to unhold this?

**Pull request merged**